plugins/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/embedded/EmbeddedEditorModelAccess.java

```diff
@@ -92,6 +108,41 @@ public void updateModel(String prefix, String editablePart, String suffix) {
 		this.viewer.getUndoManager().connect(this.viewer);
 		this.viewer.setRedraw(true);
 	}
+	
+	public void updateModel(EObject model, final EObject editablePart) {

```

Comment:
instead of passing in a editablePart as EObject, it may make sense to pass in a more generic selector instead, i.e. 

Function<XtextResource, ITextRegion> editablePartSelector

Otherwise use cases such as 
- including/excluding leading comments
- displaying the region spanning two or more EObjects
are not possible

The method updateModel(EObject model, final EObject editablePart) could be a convenience method delegating to the more general method.